File: python_playground/insertion.py
def sort(list_to_sort: list[int]) -> list[int]:
    for i in range(1, len(list_to_sort)):
        key = list_to_sort[i]
        j = i - 1
        while j >= 0:
            if list_to_sort[j] > key:
                list_to_sort[j + 1] = list_to_sort[j]
                j -= 1
            else:
                break
        list_to_sort[j + 1] = key

    return list_to_sort

File: algo_visualizer/src/algos/insertion.ts
import { BigO } from "./SortAlgorithm";
import type SortAlgorithm from "./SortAlgorithm";

export default class Insertion implements SortAlgorithm {
  readonly name = "Insertion Sort";
  readonly expectedBigO: BigO = BigO.N_SQUARED;
  readonly chartColor: string = "#10b981";

  sort = (listToSort: number[]): number[] => {
    for (let i = 1; i < listToSort.length; i++) {
      const key = listToSort[i];
      let j = i - 1;

      while (j >= 0) {
        if (listToSort[j] > key) {
          listToSort[j + 1] = listToSort[j];
          j--;
        } else {
          break;
        }
      }
      listToSort[j + 1] = key;
    }
    return listToSort;
  };
}
